omero/developers/build-system.txt

```diff
@@ -165,6 +87,106 @@ of the :model_doc:`OME-Model <ome-xml/>` and generate our |OmeroModel|.
+each :file:`ivy.xml` file and stores the resolved artifacts in a
+:ivydoc:`cache <settings/caches/cache.html>` to speed up other processes. The
+OMERO build system defines and uses two kinds of caches:
+
+#. a local cache under :file:`lib/cache`
+#. the maven cache under :file:`~/.m2/repository`
+
+.. note::
+
+   When the Ivy configuration file or the version number is changed, the
+   cache can become stale. Calling ``./build.py clean`` from the top-level
+   build will delete the content of the local cache.
+
+:ivydoc:`Resolvers <settings/resolvers.html>` are key to how Ivy functions. Multi-resolvers can be defined (as granular as for an individual jar) in order
+to pick up the latest version of any library from a
+:ivydoc:`repository <resolver/ibiblio.html>`, a
+:ivydoc:`generic URL <resolver/url.html>` or from the
+:ivydoc:`local file system <resolver/filesystem.html>`.
+
+Currently, the default resolver used by the build system, called
+``omero-resolver``, looks in the following locations by order of precedence:
+
+#. :file:`target/repository`
+#. :file:`lib/repository`
+#. :file:`target/test-repository`
+#. :file:`~/.m2/repository`
+#. `Maven central repository <http://central.sonatype.org/>`_
+#. `OME artifactory <http://artifacts.openmicroscopy.org>`_
+#. `SpringSource repository <http://repository.springsource.com/>`_
+
+.. note::
+
+   The Maven Central and OME artifactory resolvers currently have
+   `usepoms` set to `false` meaning they do not resolve transitive
+   dependencies.
+
+To define its dependencies, each component uses a top-level
+:ivydoc:`Ivy file <ivyfile.html>`, :file:`ivy.xml`, for the build and
+optionally another Ivy file, :file:`test.xml`, for the tests.
+
+The OMERO build system defines and uses four types of Ivy
+:ivydoc:`configurations <ivyfile/configurations.html>`:
+
+#. build
+#. server
+#. client
+#. test
+
+While building, most Java components follow the same lifecycle define in
+:source:`lifecycle.xml <components/antlib/resources/lifecycle.xml>`. The
+default `dist` target for each component calls each of the following steps in
+order:
+
+#. retrieve: :ivydoc:`retrieve <use/retrieve.html>` the resolved dependencies
+   and copy them under :file:`target/libs`
+#. prepare: prepare various resources (property files,
+   :source:`lib/logback-build.xml`)
+#. generate: copy all resources from the previous step for compilation
+#. compile: compile the source files into the destination repository
+#. package-extra: package the sources and the Javadoc into Jar files for
+   publication
+#. package: package the compiled classes into a Jar file for publication
+#. install: convert the component Ivy file into a pom file using
+   :ivydoc:`makepom <use/makepom.html>` and
+   :ivydoc:`publish <use/publish.html>` the component artifacts
+
+Individual components can override the content of this default lifecycle via
+their :file:`build.xml`.
+
 .. _build#OmeroTools:
 
 OmeroTools
```

Comment:
Just below this line...

"Similarly' to what? This section may have relied on an earlier section being directly above it. I'd just drop the word and start, 'The ant build...'